plugins/aggregators/histogram/histogram.go

```diff
@@ -151,6 +170,9 @@ func (h *HistogramAggregator) Add(in telegraf.Metric) {
 				index := sort.SearchFloat64s(buckets, value)
 				agr.histogramCollection[field][index]++
 			}
+			if !addTime.IsZero() {
```

Comment:
In which cases the time is zero? Do we really need that check?

Reply:
`addTime` is zero when `h.ExpirationInterval` is zero (when expiration is turned off). It allows to omit needless executing of `h.timeFun()`

Reply:
We could replace `!addTime.IsZero()` by `h.ExpirationInterval != 0` but the `!addTime.IsZero()` check allows to keep in mind only `addTime` value (reduce number of contexts)

Reply:
Please keep the `IsZero()` call and add a small comment describing that this handles the case of

> h.ExpirationInterval is zero (when expiration is turned off)

for others wanting to follow your code. :-)